src/Strategist/Strategist.cpp

```diff
@@ -0,0 +1,60 @@
+#include "Strategist.h"
+
+void Strategist::onStart() {
+    // Get initial gamestate information
+    minerals_spent = 0;
+    supply_total = BWAPI::Broodwar->self()->supplyTotal();
+    enemy_race = BWAPI::Broodwar->enemy()->getRace(); // Try to get enemy race on startup.  If the enemy chose random race, unknown will be returned.
+    chooseOpeningBuildOrder();
+}
+
+void Strategist::onFrame(){
+    
+    // Check if we need to add something to the queue
+    if (build_order_queue.empty() == false) {
+        updateUnitQueue();
+    }
+}
+
+void Strategist::incrementSupply(){
+    supply_total += 16; // supply provided by an overlord
+}
+
+void Strategist::decrementSupply(){
+    supply_total -= 16;// supply provided by an overlord, should trigger on overlord death.
+}
+
+void Strategist::DetermineMapSize(){
+    // Need to figure out how we want to determine this, or if we even want to determine it in the strategist.
+    map_size = smallest;
+}
+
+void Strategist::chooseOpeningBuildOrder(){
+    // For now, simply populate the queue.
+    for (auto it = build_order_list.begin(); it != build_order_list.end(); ++it){
```

Comment:
`build_order_list` is an `std::initializer_list`, so you can just create a new `std::queue` object passing `build_order_list` into the constructor, like:
`build_order_queue = std::queue<std::pair<BWAPI::UnitType, int>>(build_order_list);`

Since we have the explicit build order lists, `build_order_list` is redundant, just perform checks to see what build order to do and initialize `build_order_queue` by that.